Non-determinism in what language is being used to parse patterns

**Describe the bug**
I have been observing a lot of inconsistencies with how Semgrep CLI runs on local repos. I have a rule.yaml file with a couple of rules in it. It looks like below:

```
rules:
- id: hardcoded-secrets-lang
  patterns:
    - pattern-either:
      - pattern: |
          accessToken: "$A"
      - pattern: config.secret_key = $A
      - pattern: |
          AWS_SECRET_ACCESS_KEY: "$A"
  message: Don't hardcoded secrets in code. Use Vault instead.
  languages: [yaml, ruby, ts, js]
  severity: WARNING
- id: hardcoded-secrets-generic
  patterns:
    - pattern-either:
      - pattern: |
          AWS_SECRET_ACCESS_KEY=$A
  message: Don't hardcoded secrets in code. Use Vault instead.
  languages: [generic]
  severity: WARNING
```


When I run this rule on the repos (`semgrep --config rule.yaml repo`) more often than not, it complains with an error like -

```
Scanning across multiple languages:
    <multilang> | 1 rule  × 132 files
             ts | 1 rule  ×  74 files
           yaml | 1 rule  ×  13 files
             js | 1 rule  ×   3 files

    0%|                                                                                                                                                                                   |0/222 tasks
[ERROR] Pattern parse error in rule hardcoded-secrets-lang:
 Invalid pattern for TypeScript
```

Sometimes, it is `Invalid pattern for JavaScript`

Sometimes, it runs just fine and flags the things that I want it to flag.

It almost feels like the parser is flaky and cannot be relied upon to correctly identify all the matches. Is this a known bug? Am I doing something wrong?

**To Reproduce**
See description above

**Expected behavior**
The parser should successfully parse and show the findings as opposed to failing not able to determine what language it is.

**Screenshots**
N/A

**What is the priority of the bug to you?**

- [x] P0: blocking your adoption of Semgrep or workflow
- [ ] P1: important to fix or quite annoying
- [ ] P2: regular bug that should get fixed

**Environment**
Semgrep CLI, version `0.90.0` 

**Use case**
This will allow me to consistently see the same findings on running the Semgrep CLI. Right now, its inconsistent and I am not sure I can rely on the findings.

--- a/semgrep/semgrep/rule.py
+++ b/semgrep/semgrep/rule.py
@@ -61,7 +61,7 @@ class Rule:
             language == LANGUAGE.resolve("javascript") for language in rule_languages
         ):
             rule_languages.add(LANGUAGE.resolve("typescript"))
-            self._raw["languages"] = [str(l) for l in rule_languages]
+            self._raw["languages"] = sorted(str(l) for l in rule_languages)
 
         self._languages = sorted(rule_languages)
 

